src/useSpeech/index.ts

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useEffect, useState } from 'react';
+
+type SpeechOptions = {
+	voice?: SpeechSynthesisVoice;
+	rate?: number;
+	pitch?: number;
+	volume?: number;
+	lang?: string;
+};
+
+type SpeechPlayStatus = 'init' | 'play' | 'pause' | 'end' | 'error';
+
+export type SpeechState = {
+	isPlaying: boolean;
+	status: SpeechPlayStatus;
+	errorMessage?: string;
+};
+
+/**
+ * Hook for playing text using the SpeechSynthesis API.
+ *
+ * @param {string} text - Text to play.
+ * @param {ISpeechOptions} options - Options for playing the text.
```

Comment:
still having `I` prefix

Reply:
This change also been address [here](https://github.com/react-hookz/web/pull/1378/commits/58422c4b619c65faeea17e04af3e9afa4b9bdfe3)